store/fifo_store.go

```diff
@@ -0,0 +1,167 @@
+package store
+
+import (
+	"container/list"
+	"context"
+	"sync"
+	"time"
+
+	"github.com/kiali/kiali/log"
+	"github.com/kiali/kiali/util"
+)
+
+const (
+	defaultExpirationCheckInterval = 1 * time.Minute
+	defaultTTL                     = 5 * time.Minute
+)
+
+type Stats struct {
+	HitRate string `json:"hitRate,omitempty"`
+	Size    int    `json:"size,omitempty"`
+}
+
+type StoreStats struct {
+	Hits          int
+	TotalRequests int
+	Size          int
+}
+
+// FifoStore uses a FIFO approach storage and is safe for concurrent use.
+type FifoStore[K comparable, V any] struct {
```

Comment:
`FIFOStore` should implement the `Store` interface but like I said if you embed the `Store` interface in the `FIFOStore` struct then you will get all the methods you don't explicitly implement "for free".

Reply:
Actually you'll probably want the expirationStore last since when keys expire you want the order to be updated/preserved:
```
vanillaStore := store.New[string, *model.TracingSingleTrace]()
fifoStore := store.NewFIFOStore(vanillaStore, capacity)
tempoCache.Store = store.NewExpirationStore(ctx, fifoStore, nil, nil)
```